src/components/Schedule/TableView.tsx

```diff
@@ -1,18 +1,166 @@
+const splitCourseDays = (course: CourseSection) => {
+  const splitDays = course.meetingTimes.map((meetingTime) => {
+    const { startTime, endTime, day } = meetingTime;
+
+    //2022-05-31T is a monday, and we just need the base to be a monday
+    //HH:MM:SS.000-7:00 for time format, it comes in HHMM format.
+    const beginHoursMinutes = [
+      startTime.slice(0, startTime.length / 2),
+      startTime.slice(-startTime.length / 2),
+    ];
+    const endHourMinutes = [
+      endTime.slice(0, endTime.length / 2),
+      endTime.slice(-endTime.length / 2),
+    ];
+
+    const start =
+      "2022-05-31T" +
+      beginHoursMinutes[0] +
+      ":" +
+      beginHoursMinutes[1] +
+      ":00.000-07:00";
+    const end =
+      "2022-05-31T" +
+      endHourMinutes[0] +
+      ":" +
+      endHourMinutes[1] +
+      ":00.000-07:00";
+
+    const startDate = new Date(start);
+    const endDate = new Date(end);
+
+    //change the date based on what day of the week it's supposed to be
+    const dayShift = weekdayToInt(day);
+    startDate.setDate(startDate.getDate() + dayShift - 1);
+    endDate.setDate(endDate.getDate() + dayShift - 1);
+
+    const meetingDays = {
+      startDate: startDate,
+      endDate: endDate,
+    };
+
+    return meetingDays;
+  });
+
+  return splitDays;
+};
+
+//convert Course object data from backend to the structure DevExtreme expects
+const buildAppointments = (courses: CourseSection[]) => {
+  const appointments = courses.flatMap((course) => {
+    const professors = course.professors.map((prof) => prof.username); //TO-DO switch to displayName, when its in schema
+
+    const assignedCourses = splitCourseDays(course).map((meetingTime) => {
+      return {
+        courseTitle: course.CourseID.code, //not in schema yet
+        courseNumber: course.CourseID.code,
+        subject: course.CourseID.subject,
+        section: "TEMP_SECTION", //not in schema yet
+        prof: professors,
+        classSize: course.capacity,
+        startDate: meetingTime.startDate,
+        endDate: meetingTime.endDate,
+      } as Appointment;
+    });
+
+    return assignedCourses;
+  });
+  return appointments;
+};
+
+const populateTable = (courses: Appointment[])=>{
+  var table_b = document.getElementById("table_body");
+  if (table_b != null){
+    //remove previous table
+    if (table_b.childNodes.length !== 0){
+      while(table_b.childNodes.length > 0){
+        table_b.childNodes[0].remove();
+      }
+    }
+    var nTd, nTr; 
+    for (var i = 0;i < courses.length; i++){
+      nTr = document.createElement('Tr');
+
+      var sdate = courses[i].startDate.toString().split(' ').slice(0, 4).join(' ');
+      var stime = courses[i].startDate.toString().split(' ').slice(4, 5).join(' ');
+      var edate = courses[i].endDate.toString().split(' ').slice(0, 4).join(' ');
+      var etime = courses[i].endDate.toString().split(' ').slice(4, 5).join(' ')
+
+      //Course title
+      nTd = document.createElement('Td');
```

Comment:
Is there some way that we could put this into a component so we don't have the repeated code?

Reply:
Thanks! Ill work on this